### DISCUSSION TITLE
queryRaw fails while queryRawUnsafe succeeds with the same query.

### DISCUSSION THREAD

**Discussion by ddowningms** [prisma/prisma#12902] [Q&A] [answered]
I'm running Prisma on a Windows 10 computer and using Visual Studio Code to develop my API.

I'm trying to use the $queryRaw on a similar query as the following:

```
SELECT DISTINCT b.Col1,b.Col2,b.Col3,b.Col4,v.Col1
FROM [dbo].[Table1] b
JOIN [dbo].[Table2] v ON b.[KeyColumn] = v.[ForeignKeyColumn]
WHERE Col8 LIKE 'matchstring' AND b.Date > '2022-02-19T18:45:31.640Z'
ORDER BY Col1, Date

```
I'm getting the following error (***query***  is above):

`Raw query failed. Code: `203`. Message: `The name '***query***' is not a valid identifier.``

The exact same query works correctly if I use $queryRawUnsafe.

I prefer using the $queryRaw as it should prevent SQL Injection.

Any suggestions are greatly appreciated.

Thank you.

**Comment by ddowningms** [comment, marked as answer]
After a lot of trial and error, I discovered that not putting the single quotes around the date seem to resolve the issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
